homeassistant/components/mqtt/util.py

```diff
@@ -61,6 +61,14 @@ def valid_subscribe_topic(value: Any) -> str:
     return value
 
 
+def valid_subscribe_topic_template(value: Any) -> Any:
```

Comment:
```suggestion
def valid_subscribe_topic_template(value: Any) -> Union[str, template.Template]:
```

Is it a good idea to return different types, especially since Template has an optimization if it's static and so has hardly any overhead?